packages/typespec-ts/src/modular/helpers/typeHelpers.ts

```diff
@@ -1,155 +1,226 @@
 import { EnumValue, Type } from "../modularCodeModel.js";
 
+/**
+ * Represents metadata for a given Type to generate the TypeScript equivalent.
+ */
 export interface TypeMetadata {
   name: string;
   originModule?: string;
   isRelative?: boolean;
-  modifier?: "Array";
-}
-
-function getNullableType(name: string, type: Type): string {
-  if (type.nullable) {
-    return `(${name} | null)`;
-  }
-
-  return name;
+  nullable?: boolean;
+  modifier?: "Array" | "NullableArray";
```

Comment:
I am wondering if it's really neccessary to have a NullableArray special logic ?

Reply:
yup removed